### ISSUE TITLE
A number of build and runtime errors occurred during initial setup

### ISSUE THREAD

**Issue by sskorol** [introlab/opentera-webrtc-ros#52]
Hi,

I want to play with webrtc demo first. But encountered lots of issues during the setup process. Here's a list of my findings.
`catkin_make` has failed for me due to a number of missing deps: `fftw3-dev` / `libconfig-dev` and `gfortran` of a specific version.

Had to spent some time on the latter, as the trickiest part here was not the missing `gfortran` (it was installed in my system), but rather an inconsistency in `gcc` / `g++` deps. These 3 must match. But I had `gcc/g++` 8.4.0 and `gfortran` 9.4.0 installed. And the main problem here is that for my Ubuntu 20.04 distribution there was neither `gcc/g++-9.4.0` nor `gfortran-8.4.0` available. So I have to install the 10th version of these 3 and run `update-alternatives` to continue the build process.

Then it failed on the python dependencies installation. It was a weird one as it didn't have permission to install packages linked to python3.7. However, I have a 3.8 version installed and available via `python3 -V` and `update-alternatives`. I had python3.7 at some point in the past but couldn't find any direct system reference except the availability of a python3.7 folder in the `/usr/local/lib`. So I have to remove it first. And then the build failed again due to the missing python interpreter error. The root cause was here: `./opentera-webrtc-ros/opentera_webrtc_ros/opentera-webrtc/CMakeLists.txt`. I'm not sure why, but if we try to find `python 3` package, it can't see an installed 3.8 version. When I replaced 3 with 3.8 the build finally passed. So it'd probably make sense to make python version configurable via parameter or env variable.

Ok, so setup is done. But when I tried to run the following demo:
```shell
roslaunch opentera_webrtc_demos demo.launch
```
it failed with a bunch of errors related to missing `client_config.json`:
```shell
Traceback (most recent call last):
  File "/home/user/ros1_workspaces/teleop_ws/src/opentera-webrtc-ros/opentera_client_ros/scripts/opentera_client_ros.py", line 256, in <module>
    with config_file_name.open() as json_file:
  File "/usr/lib/python3.8/pathlib.py", line 1222, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
  File "/usr/lib/python3.8/pathlib.py", line 1078, in _opener
    return self._accessor.open(self, flags, mode)
FileNotFoundError: [Errno 2] No such file or directory: '/home/user/.ros/opentera/client_config.json'
```
I don't see any notes about this config in the README. So wondering where I can find it or what should be the content?

**Comment by mamaheux** [comment]
Hi,

Thank you for reporting the dependencies issues.

`client_config.json` is useful when the robot is managed by [OpenTera](https://github.com/introlab/opentera). To use the project without OpenTera, you can use the following command:
```bash
roslaunch opentera_webrtc_demos demo.launch is_stand_alone:=true
```

Then, the web interface is accessible from this URL: [http://localhost:8080/index.html#/user?pwd=abc&robot=BEAM](http://localhost:8080/index.html#/user?pwd=abc&robot=BEAM).

**Comment by sskorol** [comment]
@mamaheux thanks, `is_stand_alone:=true` works. However, I see 2 apps rendering the same stream (desktop and web). Is it intentional? Also, do you support 3rd-party integrations? I mean if I want to stream video from my own robot's camera, is it possible? Or I should create some sort of adapter? Would be greatly appreciated any references to the integration process or maybe you already have some examples.

**Comment by doumdi** [comment]
@sskorol the demo is intended to show what will be displayed on the robot screen (the Qt GUI) and from the operator (web).  Most of the time, what you will need to do is to run the signaling server and the frontend on a different server that is accessible from the Internet and connect both the robot and the web client through it.  It is always better to have a TLS certificate on the server to avoid warnings and problems.

Yes, any video stream is supported since we use standard [sensor_msgs/Image](http://docs.ros.org/en/lunar/api/sensor_msgs/html/index-msg.html) messages and encode the image with WebRTC using the [ROSStreamBridge](https://github.com/introlab/opentera-webrtc-ros/tree/main/opentera_webrtc_ros) nodes provided in our repository.  

The demo also uses the WebRTC data channel to send [geometry_msgs/Twist](http://docs.ros.org/en/noetic/api/geometry_msgs/html/msg/Twist.html). The are handled in the [RosJsonDataHandler](https://github.com/introlab/opentera-webrtc-ros/blob/main/opentera_webrtc_ros/src/RosJsonDataHandler.cpp). 

I know this is quite a complicated setup, but we will try to add documentation in the next few days.  However, the solution is quite flexible I believe it can be adapted to your needs. We have put a lot of efforts into this projet by integrating WebRTC natively with C++/Python/ROS.

We have made a [diagram](https://github.com/introlab/opentera-webrtc-ros/tree/main/drawings) to show how everything is connected.

Suggestions are always welcomed!

Regards,

Dominic

**Comment by sskorol** [comment]
@doumdi thanks for the clarification! Looking forward to seeing some more docs, as there are lots of components and cross-references in the existing READMEs. So it's hard to follow all the relationships at the moment.

From the existing launch files, I see I can disable a bunch of services like turtlebot sim, maps generator, desktop GUI, etc. However, it's not yet clear how can I access the camera. Should I create a separate node that publishes raw images from the robot's camera to a specific topic that opentera-webrtc is listening to? From web-GUI I can't enable my local camera either. It's visible in the list, but can't be activated.

**Comment by sskorol** [comment]
Anyway, it's good to see that I can at least stream my robot's camera to web-ui. Will play with settings tomorrow.

**Comment by doumdi** [comment]
@sskorol Camera problems on the web side are often due to browser security policies. With TLS enabled and a valid certificate, this should be avoided.
